### ISSUE TITLE
free draw thickness and scaling

### ISSUE THREAD

**Issue by firinael** [excalidraw/excalidraw#3693]
Two issues I've had while using Excalidraw are:

- Even the thinnest free draw option is too thick;
- Scaling down drawn objects does not scale down their thickness, and that means scaling down is almost unusable.

I'd love to have thinner line options, and for scaling to scale down thickness.

Also, this is an aside but touch input is very slow on my laptop, could this be because it has little to no VRAM? I haven't experienced slow drawing input in any other software, so it's pretty weird to see it in Excalidraw.

**Comment by locofocos** [comment]
I noticed this same issue, whenever the pencil tool was changed. I have some icons I drew with the old pencil tool (maybe 6-9 months ago), and I can't recreate them now. We've lost the ability to draw small things with detail. Notice how I can't create a legible calculator icon any more:

![image](https://user-images.githubusercontent.com/4809268/129425570-cba5613b-eb51-450e-a234-5f4761308a81.png)

I think making the thickness scale when resizing would make a ton of sense. If I wanted something with a lot of detail, I could draw it large and then resize it. Providing a smaller line size would be nice (so that I could just zoom in, draw with a thin line, and be done), but being able to draw large and then resize would accomplish the same thing.

Alternatively, having access to the legacy pencil tool would also be nice. One benefit: the legacy pencil tool has the same look and feel as all the other lines throughout excalidraw (for example, the arrow tool). Notice the difference:

![image](https://user-images.githubusercontent.com/4809268/129425827-8143377a-0b1f-4f0d-aae8-a35b74e76807.png)

**Comment by locofocos** [comment]
Maintainers, would you be interested in a contribution that makes pencil line thickness scale when resizing?

**Comment by dwelle** [comment]
@locofocos I'm not sure how we'd differentiate from a big drawing that was resized down, and a drawing that was drawn in that size to begin with. Implementation-wise you could add a `scale` property or similar (something I'm doing for the upcoming image support), but it'd also have an effect on the app dynamic in general. For contrast, we're not doing any re-scaling for other types of shapes either.

As for losing the ability to draw shapes you were able to do previously, see my comment I made just a while ago: https://github.com/excalidraw/excalidraw/issues/3916#issuecomment-898883633

**Comment by locofocos** [comment]
Thanks for the workaround! (using the line tool)

> For contrast, we're not doing any re-scaling for other types of shapes either.

That makes sense. Considering that, I think offering a thinner pencil stroke would be useful. What are your thoughts on building a 4th thickness option? I know it would be hard to please everyone, but what if we chose a thickness that more closely approximates the other lines + shapes in Excalidraw?

Offering approximately the same thickness would be nice because you could draw things that more closely resemble built in shapes. It would look and feel more cohesive IMO.

**Comment by locofocos** [comment]
After toying around with the code, I propose reducing the thickness of freedraw lines (at all stroke widths) so that freedraw lines have approximately the same thickness as other elements (of the same stroke width).

https://github.com/excalidraw/excalidraw/blob/9581c4552287ac648bbce08c82e3bd51919c4689/src/renderer/renderElement.ts#L801

Literally just change this `6` to `2`.

![image](https://user-images.githubusercontent.com/4809268/129465219-31aef397-81f6-4d78-8f77-57ad59d0ddd4.png)

![image](https://user-images.githubusercontent.com/4809268/129465229-1a41fbd6-d090-4ee4-b639-582bc38d17a4.png)

See how the freedraw lines very closely approximates the thickness of arrows and boxes? I think the freedraw lines blend in much better this way. I don't have the context of why the new freedraw lines were made so thick, but I think blending in with other elements is a more compelling use-case than offering extra thick lines.

**Comment by firinael** [comment]
I thoroughly support the above suggestion, would make the app much more useful for drawing diagrams and detailed figures in general.

That being said, the issue with line thickness staying the same despite scaling down objects is the biggest issue I can see with Excalidraw's current implementation, and it staying like that would be a very annoying and bothersome quirk.

**Comment by firinael** [comment]
@locofocos I'm just a random user, so whatever's best for those doing the actual work works for me!

**Comment by steveruizok** [comment]
When we implemented the pencil tool, we did intentionally make it a bit thicker than the other shapes, with annotations in mind. Should we reconsider?

**Comment by xixixao** [comment]
I am 100% in the camp of aligning the thickness with other shapes. If you want to use draw for annotations, change the thickness. Let's not force this use-case on everyone / remove the ability to use draw for other use cases.

**Comment by melsophos** [comment]
I completely support changing the stroke width of the `freedraw` to match the other tools, and perhaps adding more width if needed. Currently this is quite hard to get nice pictures by `freedraw`ing.

**Comment by piotrnajda3000** [comment]
It would be nice to be able to write things down with my graphic table, for example like in Smoothdraw (used in Khanacademy's videos) 

Minimum stroke width is too thick for writing comfortably, compare

Smoothdraw
![image](https://user-images.githubusercontent.com/58721991/140571748-9d08c3e0-8ab3-4932-8a05-0ec8fd760bba.png)

Excalidraw
![image](https://user-images.githubusercontent.com/58721991/140571782-49c6b47e-cf5a-4089-90a0-e8521c302385.png)

**Comment by melsophos** [comment]
Same for me, my handwriting does not produce anything recognizable when using the `freedraw` tool (which is ironical since the larger width was chosen to improve handwriting, if I understand #3500).

**Comment by mchartigan** [comment]
> After toying around with the code, I propose reducing the thickness of freedraw lines (at all stroke widths) so that freedraw lines have approximately the same thickness as other elements (of the same stroke width).
> 
> https://github.com/excalidraw/excalidraw/blob/9581c4552287ac648bbce08c82e3bd51919c4689/src/renderer/renderElement.ts#L801
> 
> Literally just change this `6` to `2`.
> 
> ![image](https://user-images.githubusercontent.com/4809268/129465219-31aef397-81f6-4d78-8f77-57ad59d0ddd4.png)
> 
> ![image](https://user-images.githubusercontent.com/4809268/129465229-1a41fbd6-d090-4ee4-b639-582bc38d17a4.png)
> 
> See how the freedraw lines very closely approximate the thickness of arrows and boxes? I think the freedraw lines blend in much better this way. I don't have the context of why the new freedraw lines were made so thick, but I think blending in with other elements is a more compelling use-case than offering extra thick lines.
> 
> I'm also really pleased with how I can recreate my little calculator icon!
> 
> ![image](https://user-images.githubusercontent.com/4809268/129465380-7b841556-1983-4654-b1d8-6e82e2ed2d69.png)
> 
> edit - This would require a data migration if we needed to avoid changing the visual thickness of existing freedraw lines. I think current lines with strokeWidth == 1 would need to be changed to strokeWidth 2. Likewise current freedraw lines with strokeWidth 2 would need to be changed to strokeWidth 4. The existing strokeWidth 4 lines would just keep their strokeWidth value (and visually become a bit thinner).
> 
> Of course, we could calculate some _exact_ strokeWidth to keep all lines exactly the same. Probably multiply them all by 3? But then people would have some previously freedraw lines at a particular width that couldn't be recreated after this proposed change, so my first data migration suggestion avoids that.

Reviving this thread cuz I'm a new user, but I think an easy solution would be to add 1 or 2 more thickness options below the current smallest line thickness, maybe so that there are 5 options total. Potentially 2 more UI buttons for stroke width = 0.25 and stroke width = 0.5. This way the old options wouldn't be lost, just new ones added

**Comment by tobei** [comment]
I fully support having the pen tool produce by default lines comparable to the built-in shapes. My pen and tablet addition to diagrams looks bold and ugly in comparison to built-in shapes as it it :)

**Comment by neurosock** [comment]
Supporting this too. Even without a stylus, combining draw and any other shape and not being able to match the widths, looks no pretty 👩🏽‍🎨

**Comment by zsviczian** [comment]
Changing this looks like a good solution, but it will have an effect on saved drawings.

https://github.com/excalidraw/excalidraw/blob/9581c4552287ac648bbce08c82e3bd51919c4689/src/renderer/renderElement.ts#L801

Adding one (width = 0.5), maybe two more presets looks like a better option. 

![image](https://user-images.githubusercontent.com/14358394/151551926-fecad7ae-8703-49af-bb5b-52db546a78b4.png)

**Comment by neurosock** [comment]
Hummm, good point @zsviczian 
but I guess 5 options would break the simplicity rules. I am not familiar with the code but keeping renderElement.ts:801 as it is and having only option 2,1, 0.5 would still keep backward compatibility, I guess??

btw @zsviczian thks for that great work you're doing with the Obsidian plugin. It is just fantastic!

**Comment by zsviczian** [comment]
> Hummm, good point @zsviczian but I guess 5 options would break the simplicity rules. I am not familiar with the code but keeping renderElement.ts:801 as it is and having only option 2,1, 0.5 would still keep backward compatibility, I guess??
> 
> btw @zsviczian thks for that great work you're doing with the Obsidian plugin. It is just fantastic!

Agree, 5 is too many. See my proposed PR with just one more stroke width preset.

**Comment by zsviczian** [comment]
> Hummm, good point @zsviczian but I guess 5 options would break the simplicity rules. I am not familiar with the code but keeping renderElement.ts:801 as it is and having only option 2,1, 0.5 would still keep backward compatibility, I guess??
> 
> btw @zsviczian thks for that great work you're doing with the Obsidian plugin. It is just fantastic!

You could manage to show 2,1,0.5 for freedraw and 1,2,4 for other elements. The issue would come when you have multiple elements selected...  which set of thicknesses do you show on the Panel?

The ideal solution would still be to change ` size: element.strokeWidth * 6` to ` size: element.strokeWidth * 2`.
Given backward compatibility, adding an extraThin line I think is good enough.

**Comment by dwelle** [comment]
I agree that the ideal solution is changing the stroke width. We should only add new tools/settings if it makes sense on its own, not as a fix for backward compat.

So we just need to figure out how to migrate.

There are 2 main options:

1. change the property name to something else (`strokeWidth` → `strokeSize`)
2. bump json schema version. Schema older than whatever the current number is (`1`?) would be migrated to previous values.

**Comment by zsviczian** [comment]
> * change the property name to something else (`strokeWidth` → `strokeSize`)

I find this the more messy option. What do you do next when a new issue is found, this time with strokeSize. Change it from strokeSize to strokeStrength? 

> * bump json schema version. Schema older than whatever the current number is (`1`?) would be migrated to previous values. One problem with this is we (wrongly) don't store schema version for cases like libraries, or when copy/pasting, which we'd need to fix as well.

I would be in favor of bumping the schema version. If you start to store schema version in copy/paste and library from now on, then you can identify legacy data because they won't have the schema version number. `const version = data.version??"2"`

BTW, the current version number is 2.

**Comment by dwelle** [comment]
Actually, as per the OP, isn't the main issue here aligning thickness of the freedraw tool to other tools? That means we wanna shift all the values (https://github.com/excalidraw/excalidraw/issues/3693#issuecomment-982919964). If we end up with a migration, we can do that.

Question is whether users also want to be able to draw thicker lines as is current behavior.

**Comment by tobei** [comment]
I struggle to imagine a usecase where user would expect a single width selected to translate in different width for every tool used. That would work different than any tool on the market. I would even like that the "handwritten" style filter applied to current shape could also be applied to manual strokes.

Shapes have 3 strokes width, line has 3 stroke widths, handwritting should have the same 3 stroke widths, i guess that's the "normal" bahaviour anyone using that kind of tool would expect, irrespective of wheter or not the handwriting tool offers 3 more widths to choose from or not.

**Comment by dwelle** [comment]
> I struggle to imagine a usecase where user would expect a single width selected to translate in different width for every tool used. That would work different than any tool on the market. I would even like that the "handwritten" style filter applied to current shape could also be applied to manual strokes.

I agree with that.

I was wondering if there's a common use case for a freedraw to also support much thicker lines.  But we can decide that later and based on user feedback. The first step will be to align the tool thickness.

**Comment by neurosock** [comment]
Conversation just reminded me that @steveruizok initial purpose with thicker lines was for them to be more visible when hand writing text https://github.com/excalidraw/excalidraw/issues/3500#issuecomment-827688251

My sincere opinion is that there are plenty of better alternative tools to draw when you have a stylus and want to fully hand draw or write. I see excalidraw as a tool mainly for desktop when you want hand draw-style-look. Would be interesting to know the stats of how many actually use it with a stylus but I guess it is hard to tell ...

I cannot think either of other use cases with very thick lines. There is always the option of having e.g. thin boxes and extra bold hand draw. Also like the 3 options.

**Comment by zsviczian** [comment]
@dwelle I am thinking of taking a slightly different approach to this issue.
Similar to stroke style, sloppiness, and edges in case of line and arrow elements, my proposal would be to implement a few pen presets.

![589469EC-D065-4307-8406-C47DFA1A510A](https://user-images.githubusercontent.com/14358394/156458862-4b229e93-7c57-4b1d-821a-4f97de1b3801.jpeg)

e.g.: fountain pen, ballpoint pen, marker, etc. 

At its simplest this could be solved by defining couple of different StrokeOption presets 

https://github.com/excalidraw/excalidraw/blob/6d0716eb6bd23685fc92277ae86dd3ea5745270f/src/renderer/renderElement.ts#L1068

I see two possible implementation approaches.
1) to define presets and store a single number in the freedraw element, similar to the fontFamily number. If the element is missing strokeOption we would default to the current StrokeOption, else to one of the presets
2) to save StrokeOptions  with every element and save to JSON. This would be my preference as this would provide full freedom/flexibility for customization, but would be slightly more complicated to implement, especially for parameters such as easing

https://github.com/steveruizok/perfect-freehand/blob/385dac18c56d609f8f9a23e7bbc4a1674190bff9/packages/perfect-freehand/src/types.ts#L19

**Comment by dwelle** [comment]
Presets should be considered as a different issue. We'll want to align the thickness either way as it doesn't make much sense for it to be inconsistent to all other shapes.

**Comment by ImkeF** [comment]
Yes, I would also very much appreciate if the stroke width of the pen would align with the other shapes.
Thanks a lot!

**Comment by kitaev-chen** [comment]
Frankly speaking, excalidraw is not suitable for pen, at least right now. There are problems everywhere, such as pen size/stroke width, smoothing, the shape and flickering of the cursor ... 

It's a terrible user experience comparing with onenote or whiteboard. 

Of course, since excalidraw is the only way to use pen in obsidian or zotero, we still look forward to its better pen mode support.

**Comment by dwelle** [comment]
The migration might be the easy part. The hard part will actually be picking the right values.

Our freedraw tool's width partly adapts to drawing speed. And our lines (perceived) thickness depends on whether you use level 0 sloppiness (architect) or level 1 sloppiness (artist).

So aligning the freedraw thickness is not easy.

Below, the freedraw thickness is kept verbatim (not multiplied by a factor at all, as we do currently).

![image](https://user-images.githubusercontent.com/5153846/204387653-e587066a-08c5-4861-9c23-4f5f859ae3a0.png)

**Comment by locofocos** [comment]
My proposal:

> Our freedraw tool's width partly adapts to drawing speed.

What if we just... turned that off? I'm just 1 person, but I'd be happier with a steady thickness. The pressure might look fancy on a marketing page, but I bet most real use cases would appreciate making slow and careful drawings that don't become overly thick in places. 

> And our lines (perceived) thickness depends on whether you use level 0 sloppiness (architect) or level 1 sloppiness (artist).

Make it line up with the default/medium sloppiness.

```ts
    size: element.strokeWidth * 2, // Match other tools
    thinning: 0.0, // Don't simulate pressure, just maintain a steady thickness
```
https://github.com/locofocos/excalidraw-opinionated-pencil/blob/6ab348918f91d76a0166f8c9427107961dbe6d8e/src/renderer/renderElement.ts#L1333-L1334

I realize this wouldn't be perfect for everyone, but I think it would be better overall, especially for most people who use the default sloppiness.

**Comment by GuangDai** [comment]
I sincerely hope there will be an option to choose stroke width. 
Although you have worries about the right width, I think you can release a beta version or something else to give users a try. Give users enough range to choose, 0.1 to 8 or 16, step 0.1. I think they will find what they need.
Can we solve the problem by ignoring the problem and not trying?

**Comment by eid0** [comment]
Let me give my feedback and leave. Hope it helps the project.
I was evaluating a tool for sketching, I was surprised with excalidraw but I consider excalidraw broken with such a thick draw stroke line and I have to consider other app for my needs, that are sketching with a few handdrawn figures with a stylus.
I'd be very surprised if i'm the only one.
Thanks

**Comment by kitaev-chen** [comment]
Another annoying thing is that the mouse in handwriting mode does not turn into a dot like a regular handwriting pen tip, but it remains a blinking cursor, which is very distracting.

**Comment by criatle808** [comment]
> Another annoying thing is that the mouse in handwriting mode does not turn into a dot like a regular handwriting pen tip, but it remains a blinking cursor, which is very distracting.

On my system (Firefox, Linux), it's a cross, which seems OK …

**Comment by criatle808** [comment]
Here is a comparison of the freehand pen (arrows, left "Hi"), shape tool (boxes), all with the thinnest line option, and the text tool with the standard font size "M". They just don't match:

![Screenshot from 2023-03-01 13-59-37](https://user-images.githubusercontent.com/93634329/222057803-7f58ca83-1a35-4d08-ba0b-193b96aa3fa4.png)

**Comment by ctwhome** [comment]
The pen is not suitable as it is now to take hand notes. It looks like it is just a matter of giving a few more size customizations than there are now. A simple slider would make it possible.

![](https://cdn.dribbble.com/users/954584/screenshots/3787724/slider.gif)

**Comment by criatle808** [comment]
I think that are two different things:

1. Acknowledge that the missing of a normal line thickness is a problem that needs to be solved.
2. Implementation details of how this is represented in the GUI.

I don't mind the representation in the GUI, as long as there is a normal line thickness to choose.

**Comment by sakaljurgis** [comment]
If you want to try out - I've added 2 changes [in my fork](https://github.com/sakaljurgis/excalidraw). The changes are:
- thin stroke toggle - so the stroke width is more or less aligned with the shapes
- pressure settings - select actual pressure from pen, simulated pressure or constant pressure. also able to delete pressure data. constant pressure setting needed changes in export file - so not readable by excalidraw - other changes exports ok
[Link to demo](https://excalidraw.sklk.lt/)

**Comment by Comprehensive-Jason** [comment]
I will look into making a PR to fix this myself by shifting the numbers as stated in https://github.com/excalidraw/excalidraw/issues/3693#issuecomment-982919964

**Comment by iamsashika** [comment]
Waiting this issue is to be solved.

**Comment by iamsashika** [comment]
Waiting for this issue to be solved.

**Comment by Comprehensive-Jason** [comment]
Adding on to https://github.com/excalidraw/excalidraw/issues/3693#issuecomment-1672136861.

For instance, we could tie the resolution of perfect-freehand to the current zoom so that lines drawn when zoomed in look just as smooth as lines drawn when zoomed out.

This would work with the freedraw thickness changes to produce a more accurate result.

**Comment by JarrettBillingsley** [comment]
Evaluating Excalidraw as a whiteboarding app, ran into this in the first 2 minutes using it, and the fact that this issue is just about to turn 3 years old is not exactly filling me with confidence

**Comment by OliverRC** [comment]
It's a shame to stumble upon this issue and see after years it remains unresolved.
There is a noticeable discrepancy in line thickness between the lines, shapes and the draw tool. Even with very fast mouse drawing the lines free hand are significantly thicker than the line and shape tools.

It renders the draw tool basically useless. Surely there is some practical resolution even considering migration of existing drawings here. 

Personally the draw tool's sensitivity to speed is way off. Even at max speed on a sensitive mouse the lines are super thick. Go slowly and they are absolutely massive.

**Comment by dmitry-markin** [comment]
> The migration might be the easy part. The hard part will actually be picking the right values.
> 
> Our freedraw tool's width partly adapts to drawing speed. And our lines (perceived) thickness depends on whether you use level 0 sloppiness (architect) or level 1 sloppiness (artist).
> 
> So aligning the freedraw thickness is not easy.
> 
> Below, the freedraw thickness is kept verbatim (not multiplied by a factor at all, as we do currently).
> 
> ![image](https://user-images.githubusercontent.com/5153846/204387653-e587066a-08c5-4861-9c23-4f5f859ae3a0.png)

@dwelle Are there still plans to align the thickness of freedraw with other tools? I would vote 100% for it, as otherwise I have to patch `renderElement.ts` to use Excalidraw with a drawing tablet.

As for the different thickness of other tools in architect and artist modes, I would like to propose the same toggle for freedraw. This way you can make the thickness not depend on the drawing speed in the architect mode and align it 1:1 with the thickness of other tools in the architect mode. One can still use the pen pressure to change the thickness, though. And, in the artist mode, you can align the thickness with other tools in the artist mode, and make it depend on the drawing speed.

**Comment by JamMaster1999** [comment]
Are there any plans to address this? There seems to be lots of great ideas in the thread but no implementation? Tldraw solved this ages ago. There is clearly enough interest to perhaps sponsor the issue?

**Comment by AliakseiKaniaveha** [comment]
Seriously, even keeping all the complexity related to scaling aside, is there any reason to have shapes and free draw thickness misaligned? What is the reason on 100% scale with the lowest thickness for each tool it looks like that:
![Image](https://github.com/user-attachments/assets/354675a2-e1ef-4706-b8e4-8ae7e9c4ea51)

Which effectively kills the beautiful ability to combine shapes with stylus-based fine-grained details and forces people to seek alternatives (that's how it look in Miro):
![Image](https://github.com/user-attachments/assets/136aa2ca-3d78-4fd0-b50a-b7c56e437f13)

**Comment by krzemyczek** [comment]
for things like this i want to learn how to code... It's a shame that i cannot help in this...

**Comment by kyle-tennison** [comment]
This seems like such a simple thing to fix, yet no action for 4 years? Hopefully someone can make a plugin that resolves this.

**Comment by JamMaster1999** [comment]
This is why I have moved on from excalidraw to tldraw. It is much better at this.

**Comment by iamsashika** [comment]
This is the whole reason why I moved to tldraw

**Comment by kitaev-chen** [comment]
> This is the whole reason why I moved to tldraw

For drawing tablet users, tldraw is worse to write with than Excalidraw... 
Of course, I am also not satisfied with Excalidraw; I can only handwrite formulas or draw in OneNote and then take screenshots to insert them into Obsidian for use.

**Comment by JamMaster1999** [comment]
That does not match my experience using an iPad. Tldraw is close to the native experience while excalidraw constantly stutters and produces jagged shapes.

Best Regards,


Founder, CEO

Sent with Spark

**Comment by kitaev-chen** [comment]
OneNote is clear, slim and stiff touch, others are unclear, obese and soft touch. Here is my test:

**Excalidraw**
![Image](https://github.com/user-attachments/assets/46e290f9-55e1-49af-9f05-2e1dba922c17)
![Image](https://github.com/user-attachments/assets/120ce4a0-df22-497a-bc26-79125640c75f)

**Tldraw**
![Image](https://github.com/user-attachments/assets/ed7e2cfa-047d-4082-9235-c98a821feddc)
![Image](https://github.com/user-attachments/assets/054b4529-b166-4ef6-906b-a84865f6a12d)

**OneNote**
![Image](https://github.com/user-attachments/assets/f5378ba7-1f82-4c00-9e1a-82c6dc1b6ee0)
![Image](https://github.com/user-attachments/assets/f5ccc472-cee9-41ad-a302-b829e0e1d703)

**Comment by JamMaster1999** [comment]
It seems like you are using these tools inside of a wrapper of some sort. 
When using tldraw directly it looks better than excalidraw. 
Of course One Note looks the best but that is not open source. 

![Image](https://github.com/user-attachments/assets/b390d522-ae18-419c-933c-da90021b8795)

![Image](https://github.com/user-attachments/assets/6a4edfd6-2fcb-4ffd-949f-4ff1d36c15eb)

**Comment by einarpersson** [comment]
Hi, new Excalidraw+ user (for our organization). This is one thing that buggers us the most. I just did a drawing and then wanted to scale it down to match another drawing but then it became just a black blob more or less. It wasn't the UX i expected.

This issue seems to resonate with a lot of people, is there nothing you can do? (now as a paying customer I feel I can whine a little ;) )

**Comment by criatle808** [comment]
I'd like to understand the reasons why this hasn’t been implemented yet. Is it due to:

- technical challenges (e.g., backward compatibility, complexity of the changes),
- limited time or lower prioritization by the maintainers,
- or other factors?

It would be helpful to have a clear consensus on this, so users can better understand the status and contributors can more effectively address the obstacles.

**Comment by Mrazator** [comment]
Hi guys, we have this one on top of our list (with P1 priority), but as you can see just the itself backlog is huge and we usually end up postponing this due to other priorities. We shall work on it ASAP, but atm I don't have an ETA that I could give you.

In the meantime, feel free to give it a shot, perhaps at least partially identifying the options and possible solutions.

**Comment by nGrave** [comment]
https://github.com/excalidraw/excalidraw/pull/9416/commits/1c0c6adbf3be31ca58be8d1b1058ee8b923161cf

![Image](https://github.com/user-attachments/assets/ce37f653-eb0d-4e76-a1b7-6e83b1975bdd)

Thoughts on the above PR  ? 


Adding this "Fine" option provides users with a much-needed thinner line option across all tools *now*, without breaking backward compatibility for existing drawings.

Ideally, stroke widths might be visually consistent across all element types *at all sizes*. However, achieving that for the existing standard stroke widths (1, 2, 4) would require changing the current rendering logic (e.g., the `4.25` multiplier for freehand), which would alter the appearance of countless existing canvases – a significant breaking change best addressed separately.

While other solutions like custom thickness sliders would be great, this discrete "Fine" option aims to be the least intrusive way to solve the immediate need for a thinner line option now without needing decide now on "simplicity" vs "flexibility" design principles and all that it entails, personally I'd like to see all thicknesses the same across all elements and could potentially just apply it to new canvases going forward.

**Comment by Pufferknutscher** [comment]
> After toying around with the code, I propose reducing the thickness of freedraw lines (at all stroke widths) so that freedraw lines have approximately the same thickness as other elements (of the same stroke width).
> 
> [excalidraw/src/renderer/renderElement.ts](https://github.com/excalidraw/excalidraw/blob/9581c4552287ac648bbce08c82e3bd51919c4689/src/renderer/renderElement.ts#L801)
> 
> Line 801 in [9581c45](/excalidraw/excalidraw/commit/9581c4552287ac648bbce08c82e3bd51919c4689)
>  size: element.strokeWidth * 6, 
> 
> Literally just change this `6` to `2`.
> 
> ![image](https://user-images.githubusercontent.com/4809268/129465219-31aef397-81f6-4d78-8f77-57ad59d0ddd4.png)
> 
> ![image](https://user-images.githubusercontent.com/4809268/129465229-1a41fbd6-d090-4ee4-b639-582bc38d17a4.png)
> 
> See how the freedraw lines very closely approximate the thickness of arrows and boxes? I think the freedraw lines blend in much better this way. I don't have the context of why the new freedraw lines were made so thick, but I think blending in with other elements is a more compelling use-case than offering extra thick lines.
> 
> I'm also really pleased with how I can recreate my little calculator icon!
> 
> ![image](https://user-images.githubusercontent.com/4809268/129465380-7b841556-1983-4654-b1d8-6e82e2ed2d69.png)
> 
> edit - This would require a data migration if we needed to avoid changing the visual thickness of existing freedraw lines. I think current lines with strokeWidth == 1 would need to be changed to strokeWidth 2. Likewise current freedraw lines with strokeWidth 2 would need to be changed to strokeWidth 4. The existing strokeWidth 4 lines would just keep their strokeWidth value (and visually become a bit thinner).
> 
> Of course, we could calculate some _exact_ strokeWidth to keep all lines exactly the same. Probably multiply them all by 3? But then people would have some previously freedraw lines at a particular width that couldn't be recreated after this proposed change, so my first data migration suggestion avoids that.

I'd like to revive this thread as I have run into this issue since using Excalidraw. This comment above seemingly shows an almost-perfect solution? The least that could be done for the end user would maybe be allowing this value to be changed in the settings or just as a variable in the data.json? Since I don't want to have to change the pen thickness each time I switch between pen and form, I will have to get into a Java IDE to open the main.js to find this value somewhere, then somehow recompile and hope nothing goes wrong, which is really inconvenient. Don't get me wrong, I'm really glad that this plugin exists at all, but it seems to me that this could be a good and mostly easy way to avoid this issue. 
If I've gotten something wrong, please correct me. 
Edit: Added a sentence.
Edit 2: I am on a Surface 7 Pro, using Win11 Obsidian with the Excalidraw Plugin. Everything is up-to-date as of today. I'm using the Surface Pen, which has a windows-enabled pressure sensor. 
Edit 3: I've seen that there was a proposed solution with the 4th pen thickness option in the GUI, however I can't see whether this was actually published and even if it was, I still have this issue on Windows. 

Edit 4: It is now 2 days later, and Ive found the custom pen option while foresting through the settings. This is an even better solution to what I had proposed.

**Comment by BH2WFR** [comment]
+1, I think pen width is too thick.

**Comment by diegomarzaa** [comment]
+1 in having the width of both freehand and figures be consistent. Don't see any reason for having them different.

For now, for other users, here's the automated fix for the current excalidraw version. Compiling excalidraw locally and then pasting the results into the plugin folder.


### Step 1: Navigate to the plugin directory after cloning this repository.

### Step 2: Install dependencies (if needed)
```bash
npm install
```

### Step 3: Apply the fix to the development file
```bash
sed -i 's/size: element\.strokeWidth \* 4\.25/size: element.strokeWidth * 1.5/g' node_modules/@zsviczian/excalidraw/dist/excalidraw.development.js
```

### Step 4: Apply the fix to the production file
```bash
sed -i 's/4\.25/1.5/g' node_modules/@zsviczian/excalidraw/dist/excalidraw.production.min.js
```

### Step 5: Verify the changes
```bash
# Should output "size: element.strokeWidth * 1.5" twice
grep -o "size: element.strokeWidth \* [0-9.]*" node_modules/@zsviczian/excalidraw/dist/excalidraw.development.js

# Should output 0 (no more 4.25 in the file)
grep -c "4\.25" node_modules/@zsviczian/excalidraw/dist/excalidraw.production.min.js
```

### Step 6: Build the plugin
```bash
npm run build:all
```

### Step 7: Copy to Obsidian plugins folder
```bash
cp dist/* <ObsidianFolderPath>/.obsidian/plugins/obsidian-excalidraw-plugin/
```

### Step 8: Restart Obsidian
Close and reopen Obsidian (or reload the Excalidraw plugin) to load the new version.

### Results for the same widths:

<img width="804" height="571" alt="Image" src="https://github.com/user-attachments/assets/8bababf9-9ca4-435d-8709-db3a49e976f7" />

**Comment by RDust4** [comment]
I've written a script that applies this patch automatically. Just make sure you call it while your pwd is inside your target vault or a subfolder.
It's not a very clean way of doing it, but it works for now.

```bash
#!/usr/bin/env bash

set -euo pipefail


TARGET=".obsidian"
VAULTROOT="$PWD"

while [[ ! -d "$VAULTROOT/$TARGET" ]]; do
    if [[ "$VAULTROOT" == "/" ]]; then
        echo "Error: Reached filesystem root without finding '$TARGET'." >&2
        exit 1
    fi
    VAULTROOT="$(dirname "$VAULTROOT")"
done

echo "Found Vault root at: $VAULTROOT"


TMP=$(mktemp -d)
echo "Working in temp dir: $TMP"
cd "$TMP"


git clone https://github.com/zsviczian/obsidian-excalidraw-plugin
cd obsidian-excalidraw-plugin


LATEST_TAG=$(git tag --sort=-v:refname | grep -E "^[0-9]+\.[0-9]+\.[0-9]+$" | head -n 1)
echo "Checking out tag: $LATEST_TAG"
git checkout "$LATEST_TAG"


echo "Installing dependencies..."
npm install


echo "Patching Excalidraw render width..."
sed -i 's/size: element\.strokeWidth \* 4\.25/size: element.strokeWidth * 0.75/g' node_modules/@zsviczian/excalidraw/dist/excalidraw.development.js
sed -i -E 's/size:4\.25\*([a-zA-Z])\.strokeWidth/size:0.75*\1.strokeWidth/g' node_modules/@zsviczian/excalidraw/dist/excalidraw.production.min.js


echo "Building plugin..."
npm run build:all


PLUGIN_DIR="$VAULTROOT/.obsidian/plugins/obsidian-excalidraw-plugin/"


if [[ -d "dist" ]]; then
    echo "Build successful. Installing to $PLUGIN_DIR..."
    mkdir -p "$PLUGIN_DIR"
    cp -r dist/* "$PLUGIN_DIR"
else
    echo "Error: Build failed. dist folder not found." >&2
    exit 1
fi


cd "$VAULTROOT"
rm -rf "$TMP"

echo "Done. Please restart Obsidian to reload the plugin."
```

**Comment by msanjayachar** [comment]
Using zoom to help reduce the stroke width to my liking worked for me.

**Comment by bjornasm** [comment]
> Using zoom to help reduce the stroke width to my liking worked for me.

Yes, I find drawing at 33% zoom works best, but it messes up the scaling a bit.

**Comment by dwelle** [comment]
> Even the thinnest free draw option is too thick;

first part of the request fixed in https://github.com/excalidraw/excalidraw/pull/11507

(took a bit long - sorry!)

> Scaling down drawn objects does not scale down their thickness, and that means scaling down is almost unusable.

scaling based on zoom is not currently planned, but we may still do it at some point. I opened a new issue for this: https://github.com/excalidraw/excalidraw/issues/11557
